projects/driver_controls/inc/event_arbiter.h

```diff
@@ -8,30 +8,34 @@
 // not pressed, which would be dangerous for the driver.
 
 #include "fsm.h"
+#include "objpool.h"
 #include "status.h"
 
 // Arbitrary FSM cap
 #define EVENT_ARBITER_MAX_FSMS 10
 
-typedef struct EventArbiterOutputData {
-  uint8_t id;
-  uint8_t state;
-  uint16_t data;
-} EventArbiterOutputData;
-
-// Typedef for the FSM arbitration functions
+// Returns whether the given event should be processed by any FSMs
 typedef bool (*EventArbiterCheck)(const Event *e);
 
-typedef void (*EventArbiterOutput)(EventArbiterOutputData data);
+typedef struct EventArbiter {
+  FSM *fsm;
+  EventArbiterCheck event_check_fn;
+} EventArbiter;
```

Comment:
Technically I think I've flipped the naming scheme here. The arbiter should be the overarching referee while these are more like rules